### DISCUSSION TITLE
Filling a geometric shape (cone)

### DISCUSSION THREAD

**Discussion by baharayhan** [Computational-Mechanics-Material-Models/chrono-mechanics#6] [Q&A] [answered]
I want to fill the cone with granular material. 
Is there any method/function for this?

**Comment by LeiShenHHU** [comment, marked as answer]
Please find the running file for a slamp test in the attachments. The modification locates at Line 166-186. 
Methods in brief:
1-Generate a big volume to cover the target specimen. "std::vector<chrono::ChVector<float>> body_points;"
2-Check one-by-one if the points is within the target specimen (any shape we want).
3-Store the effective points in the "std::vector<chrono::ChVector<float>> body_points_new"
4-Replace the "body_points" with " body_points_new" in the line 186.

<img width="1031" alt="1" src="https://github.com/Concrete-Chrono-Development/chrono-concrete/assets/120641987/6f15757d-6ae0-4322-97fb-9f32eb240a9f">
